rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -185,6 +188,88 @@ __lockless_has_parameter(
   return parameters.find(name) != parameters.end();
 }
 
+// see https://en.cppreference.com/w/cpp/types/numeric_limits/epsilon
+#define ULP 100
+RCLCPP_LOCAL
+bool
+__are_doubles_equal(double x, double y, int ulp = ULP)
+{
+  return std::abs(x - y) <= std::numeric_limits<double>::epsilon() * std::abs(x + y) * ulp;
+}
+
+RCLCPP_LOCAL
+bool
+__check_parameter_value_in_range(
+  const rcl_interfaces::msg::ParameterDescriptor & descriptor,
+  const rclcpp::ParameterValue & value)
+{
+  if (!descriptor.integer_range.empty() && value.get_type() == rclcpp::PARAMETER_INTEGER) {
+    int64_t v = value.get<int64_t>();
+    auto integer_range = descriptor.integer_range.at(0);
+    if ((v < integer_range.from_value) || (v > integer_range.to_value)) {
+      return false;
+    }
+    if (v == integer_range.to_value) {
```

Comment:
Both bounds `to_value` and `from_value` [are valid](https://github.com/ros2/rcl_interfaces/blob/master/rcl_interfaces/msg/IntegerRange.msg#L14-L22).

Reply:
Sere. It success later in one of these two places:
https://github.com/ros2/rclcpp/blob/ac852902ae2acc84c328de3661202f732ea7180a/rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp#L215-L220

But it's maybe clearer to do it in the way you propose.